tests/test_assert.cpp

```diff
@@ -14,7 +21,12 @@ void TestFail(const char *expval, const char *val, const char *exp,
   TEST_OUTPUT_LINE("TEST FAILED: %s:%d, %s in %s", file, line, exp,
                    func ? func : "");
   testing_fails++;
-  assert(0);  // assert on first failure under debug
+
+  // hook can cancel assert if return true
+  auto terminate = !global_fail_hook ||
```

Comment:
Please explain what is going on here and why this is need. @sutambe who started this file.